narwhal/primary/src/core.rs

```diff
@@ -552,10 +569,18 @@ impl Core {
                 received: header.epoch
             }
         );
+        // Even if a certificate can be created from this header, it will never get included
+        // in this node's DAG.
         ensure!(
             self.gc_round < header.round,
             DagError::TooOld(header.id.into(), header.round, self.gc_round)
         );
+        // The header round is too high for this node, which is unlikely to acquire all
+        // parent certificates in time.
+        ensure!(
+            self.highest_processed_round + MAX_HEADER_ROUND_CATCHUP_THRESHOLD > header.round,
+            DagError::TooNew(header.id.into(), header.round, self.gc_round)
+        );
```

Comment:
Great that we are thinking about the same issue! I don't think this will be a liveness issue because the node that received this advanced header cannot vote on it anyway. Will send this out as a separate PR.